### ISSUE TITLE
CA1721 should not raise on overrides

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#2914]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.6 (Latest)

#### Diagnostic ID

Example: [CA1721](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1721-property-names-should-not-match-get-methods)

#### Repro steps

```csharp
public class MyBase
{
    public object Content { get; }
    public virtual object GetContent() => null;
}

public class Foo
{
	// Issue is raised here but as I cannot change anything in here
    // there should be no issue raised.
    public override object GetContent() => this;
}
```

#### Expected behavior

No CA1721 is raised on `Foo.GetContent()` as there is no way to solve the issue except editing the base class which might not be under our control (cf `RadOpenFolderDialog` from Telerik).

#### Actual behavior

The issue is raised on `Foo.GetContent()`

I will try and have a PR fixing this issue if you are fine with it.
